scripts/tests/test-gate-selection.sh

```
@@ -0,0 +1,467 @@
+  if grep -Fq '\<' "$portable_script" || grep -Fq '\>' "$portable_script"; then
+    fail "helper uses GNU-only grep word boundaries: ${portable_script#"$repo_root/"}"
+  fi
+done
+
+default_root_recipe=$(make -s -C "$repo_root" -n test-integration)
+assert_not_contains "$default_root_recipe" "-run '^TestIntegration'" "root default integration recipe"
+assert_contains "$default_root_recipe" "./components/... ./pkg/... ./tests/..." "root integration scope"
+
+reported_root_recipe=$(make -s -C "$repo_root" -n test-integration \
+  GOTESTSUM=gotestsum TEST_REPORTS_DIR=reports/ci)
+assert_contains "$reported_root_recipe" 'reports/ci/integration-events/$package_index.json' \
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)